From 4b6d80b4a5d5d552a8ff8128d35e1df41cf5349e Mon Sep 17 00:00:00 2001
From: Li Jun <B47624@freescale.com>
Date: Mon, 22 Sep 2014 16:19:59 +0800
Subject: [PATCH 0088/5242] MLK-11340-39 usb: chipidea: otg: clear b_bus_req
 when vbus is off

commit  a75f23f0c0ae84157c3c18f6e33319cdbb3228a3 from
https://source.codeaurora.org/external/imx/linux-imx.git

In case of b_peripheral --> b_wait_acon --> b_idle due to vbus off
in b_wait_acon state, b_bus_req cannot be cleared in b_idle state,
which result in b device will do data pulse because b_bus_req is set.
This patch fix this issue by clear the input variable b_bus_req when
vbus is off.

Acked-by: Peter Chen <peter.chen@freescale.com>
Signed-off-by: Li Jun <b47624@freescale.com>
(cherry picked from commit bc600546bf9193f1a39186ad4c07a5fd497c7bfd)
(cherry picked from commit c0ea9bfcd948e2d79ea77d22756550f498281cbb)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/chipidea/otg_fsm.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/usb/chipidea/otg_fsm.c b/drivers/usb/chipidea/otg_fsm.c
index e2ccd2f..1b8d589 100644
--- a/drivers/usb/chipidea/otg_fsm.c
+++ b/drivers/usb/chipidea/otg_fsm.c
@@ -792,6 +792,8 @@ irqreturn_t ci_otg_fsm_irq(struct ci_hdrc *ci)
 				fsm->b_sess_vld = 0;
 				if (fsm->id)
 					ci_otg_add_timer(ci, B_SSEND_SRP);
+				if (fsm->b_bus_req)
+					fsm->b_bus_req = 0;
 			}
 		} else if (otg_int_src & OTGSC_AVVIS) {
 			hw_write_otgsc(ci, OTGSC_AVVIS, OTGSC_AVVIS);
-- 
1.7.9.5

